### PR TITLE
Flag gate multi-subnet cluster changes related to NEG CR.

### PR DESCRIPTION
**Pull request by sawsa307**
Flag gate changes in #2604 and #2615. Ensure no behavior changes unless MultiSubnetClusterPhase1 is enabled.

/assign @gauravkghildiyal 
/assign @swetharepakula

### CODE REVIEW

File: pkg/neg/syncers/transaction_test.go [hunk 1]
```diff
@@ -1327,10 +1327,10 @@ func TestTransactionSyncerWithNegCR(t *testing.T) {
...
 }
 
-// TestUpdateInitStatus iterates over different zone transition situation, and
-// check if NEG Object Reference in the corresponding zone has the expected State.
-func TestUpdateInitStatus(t *testing.T) {
-	t.Parallel()
+// TestUpdateInitStatusWithMultiSubnetCluster iterates over different zone
+// transition situation, and checks if NEG Object Reference in the corresponding
+// zone has the expected State.
+func TestUpdateInitStatusWithMultiSubnetCluster(t *testing.T) {
```
Comment (sawsa307) [inline pkg/neg/syncers/transaction_test.go:1333]: This is a newly added test which is MultiSubnetCluster-specific.

**Review by gauravkghildiyal — COMMENTED** [review]
Looks good. Thanks!

I've held a different opinion previously but recently come on the side of not plumbing flags through all the functions. Each new addition and subsequent cleanup of the flag ends up touching multiple function signatures and call chains. Unless we want to pass two independent values of the flag in production (not talking about tests), it's worth avoiding this mess. (That combined with not running unit-tests in parallel should not pose problems)

File: pkg/neg/manager.go [hunk 2]
```diff
@@ -647,7 +648,7 @@ func (manager *syncerManager) processNEGDeletionCandidate(svcNegCR *negv1beta1.S
 	// Since no more NEG deletion will be happening at this point, and NEG
 	// CR will not be deleted, clear the reference for deleted NEGs in the
 	// NEG CR.
-	if len(deletedNegs) != 0 {
+	if flags.F.EnableMultiSubnetClusterPhase1 && len(deletedNegs) != 0 {
```
[the remark this answers is not in the document]
Comment (sawsa307) [reply pkg/neg/manager.go]: Updated.

**Comment by sawsa307** [comment]
> Looks good. Thanks!
> 
> I've held a different opinion previously but recently come on the side of not plumbing flags through all the functions. Each new addition and subsequent cleanup of the flag ends up touching multiple function signatures and call chains. Unless we want to pass two independent values of the flag in production (not talking about tests), it's worth avoiding this mess. (That combined with not running unit-tests in parallel should not pose problems)

Yes that makes total sense.
I suppose you are talking about "not passing flag to NewNEGLinker". Just updated so it gets flag from the global var.

**Review by sawsa307 — COMMENTED** [review]
Updated.

**Comment by gauravkghildiyal** [comment]
Thanks you!

/lgtm

**Pull request merged**